src/Shared/Components/DeploymentConfigDiff/DeploymentConfigDiffMain.tsx

```diff
@@ -0,0 +1,154 @@
+    isLoading,
+    headerText = 'Compare With',
+    configList = [],
+    scrollIntoViewId,
+    selectorsConfig,
+    sortOrder,
+    onSortBtnClick,
+}: DeploymentConfigDiffMainProps) => {
+    // STATES
+    const [expandedView, setExpandedView] = useState<Record<string | number, boolean>>({})
+
+    // REF
+    const scrollRef = useRef<HTMLDivElement>()
+
+    useEffect(() => {
+        if (scrollRef?.current) {
+            scrollRef.current.scrollIntoView({ block: 'start', behavior: 'smooth' })
+        }
+    }, [scrollRef, configList])
+
+    const handleAccordionClick = (id: string) => () => {
+        setExpandedView({
+            ...expandedView,
+            [id]: !expandedView[id],
+        })
+    }
+
+    const renderHeaderSelectors = (list: DeploymentConfigDiffSelectPickerProps[]) =>
+        list.map((configItem, index) => {
+            if (configItem.type === 'string') {
+                return (
+                    <Fragment key={configItem.id}>
+                        {typeof configItem.text === 'string' ? (
+                            <p className="m-0 cn-9 fs-13 lh-20 fw-6">{configItem.text}</p>
+                        ) : (
+                            configItem.text
+                        )}
+                    </Fragment>
+                )
+            }
+
+            const { selectPickerProps } = configItem
+            return (
+                <Fragment key={configItem.id}>
+                    <div className="dc__mxw-300">
+                        <SelectPicker {...selectPickerProps} />
+                    </div>
+                    {index !== list.length - 1 && <span className="cn-9 fs-13 lh-20">/</span>}
+                </Fragment>
+            )
+        })
+
+    const renderSortButton = () => (
+        <div className="dc__border-left p-12 h-100">
+            <button
+                type="button"
+                className="dc__unset-button-styles flexbox dc__align-items-center dc__gap-6"
+                onClick={onSortBtnClick}
+            >
+                <ICSortArrowDown
+                    className="fcn-7 rotate"
+                    style={{ ['--rotateBy' as string]: sortOrder === SortingOrder.ASC ? '0deg' : '180deg' }}
+                />
+                <span className="cn-7 fs-13 lh-20 fw-6">Sort keys</span>
+            </button>
+        </div>
+    )
+
+    const renderDiffs = () =>
+        configList.map(({ id, isDeploymentTemplate, primaryConfig, secondaryConfig, title, hasDiff }) => {
+            const { heading: primaryHeading, list: primaryList } = primaryConfig
+            const { heading: secondaryHeading, list: secondaryList } = secondaryConfig
+
+            return (
+                <DeploymentConfigDiffAccordion
+                    key={`${id}-${title}`}
+                    ref={scrollIntoViewId === id ? scrollRef : null}
+                    id={id}
+                    title={title}
+                    isExpanded={!expandedView[id]}
```

Comment:
Please confirm if this check is correct.
```suggestion
                    isExpanded={expandedView[id]}
```